src/components/backup/RestoreCloudStep.tsx

```diff
@@ -130,34 +145,50 @@ export default function RestoreCloudStep() {
 
       if (status === RestoreCloudBackupResultStates.success) {
         // Store it in the keychain in case it was missing
-        await saveBackupPassword(password);
+        const hasSavedPassword = await getLocalBackupPassword();
+        if (!hasSavedPassword) {
+          await saveLocalBackupPassword(password);
+        }
```

Comment:
@skylarbarrera does this make sense to do here? This is the new local backup function we wrote not sure if you remember.

Reply:
this seems fine, i think the issue was that password could be undefined/empty? that issue seems diff